cocos/2d/CCNode.cpp

```diff
@@ -2233,4 +2175,16 @@ int Node::getAttachedNodeCount()
     return __attachedNodeCount;
 }
 
+void Node::setProgramState(backend::ProgramState* programState)
+{
+    CC_SAFE_RELEASE(_programState);
+    _programState = programState;
+    CC_SAFE_RETAIN(programState);
```

Comment:
should retain programState first, or if programState equals _ programState, then may have problem

Reply:
or should check if programState equals _ programState